soc/arm/st_stm32/common/soc_gpio.c

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2018 Song Qiang <songqiang1304521@gmail.com>
+ *
+ * Derived from scattered soc_gpio.c files under soc/st_stm32/stm32f*.
+ * Originally written by:
+ * Open-RnD Sp. z o.o. (stm32l4/soc_gpio.c)
+ * BayLibre, SAS (stm32l4/soc_gpio.c)
+ * Endre Karlson <endre.karlson@gmail.com> (stm32l0/soc_gpio.c)
+ * Yurii Hamann (stm32f7/soc_gpio.c)
+ * Linaro Limited (stm32f4/soc_gpio.c)
+ * RnDity Sp. z o.o. (stm32f3/soc_gpio.c, stm32f0/soc_gpio.c)
+ * qianfan Zhao <qianfanguijin@163.com> (stm32f2/soc_gpio.c)
+ * Open-RnD Sp, z o.o. (stm32f1/soc_gpio.c)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief
+ *
+ * Common manipulation functions for GPIO of
+ * stm32f0/f1/f2/f3/f4/f7/l0/l4 series.
+ *
+ */
+
+#include <errno.h>
+
+#include <gpio.h>
+#include <gpio/gpio_stm32.h>
+#include <soc.h>
+#include "soc_gpio_registers.h"
+#include "soc_syscfg_common.h"
+
+/* F1 series STM32 use AFIO register to configure interrupt generation
+ * and pinmux.
+ */
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+#include "afio_registers.h"
+#else
+#include "syscfg_registers.h"
+#endif
+
+int stm32_gpio_flags_to_conf(int flags, int *pincfg)
+{
+	int direction = flags & GPIO_DIR_MASK;
+	int pud = flags & GPIO_PUD_MASK;
+
+	if (!pincfg) {
+		return -EINVAL;
+	}
+
+	if (direction == GPIO_DIR_OUT) {
+#ifdef CONFIG_SOC_SERIES_STM32F1X
```

Comment:
It would be clearer to define a smart set of headers and remove all this ifdefery from this function
```suggestion
#ifdef CONFIG_SOC_SERIES_STM32F1X
#define PIN_MODE_OUTPUT STM32_MODE_OUTPUT | STM32_CNF_GP_OUTPUT | STM32_CNF_PUSH_PULL
#define PIN_MODE_INPUT ...
#define PIN_PUPD_PULLUP ...
...
#else
#define PIN_MODE_OUTPUT   STM32_MODER_OUTPUT_MODE
#define PIN_MODE_INPUT ...
#define PIN_PUPD_PULLUP ...
```

Reply:
I just finished implementing these functions with LL library. 
I was just want to ask should I push only one commit using the LL library or two commits that the first one merge separate soc_gpio.c to one file, and the second commit change the implementation to use the LL library because one commit should do only one thing?

Reply:
I have a f103zet6 based board and a f407zgt6 based board. Usually I test on both of them once I make some changes.

Reply:
> I just finished implementing these functions with LL library.
I was just want to ask should I push only one commit using the LL library or two commits that the first one merge separate soc_gpio.c to one file

Ok, great thanks. Please split into 2 commits, it will be easier to review.

Reply:
>  f103zet6 based board and a f407zgt6

Ok, thanks. I'll do some additional tests on the boards I own.